### ISSUE TITLE
MOC Catalogue Broken Indexing

### ISSUE THREAD

**Issue by astrodavid10** [WorldWideTelescope/wwt-website#279]
When selecting any image from the MOC Catalogue or the "Mars Orbiter Catalogue Imagery" It seems that the indexing for the tiles is broken in some way (web and Windows). These are found under the explorable collections and the attached images give an idea of how the indexing changes as tiles load.
![SavedView](https://user-images.githubusercontent.com/65675418/105954012-b7eb8500-6039-11eb-82ef-75008bf75b91.png)
![SavedView1](https://user-images.githubusercontent.com/65675418/105954022-bb7f0c00-6039-11eb-8c52-b582e248db27.png)
![SavedView2](https://user-images.githubusercontent.com/65675418/105954032-bf129300-6039-11eb-938f-a8a68151f2e6.png)

**Comment by pkgw** [comment]
Um, hmm. From my browser's debugger, I see that this URL can generate a bad image:

```
http://cdn.worldwidetelescope.org/wwtweb/MarsMoc.aspx?q=4,0,5&v=2
```

But if I download it manually, the image is fine. That's even with the CDN layer — at first I thought maybe the CDN cache was messed up, but it's looking like not.

**Comment by pkgw** [comment]
OK, well, I guess the browser is pickier about PNGs or something? If I just open that URL in my browser directly, it looks janky. If I open the `www.wordwidetelescope.org` version, it's fine. So maybe a CDN purge will fix this.

**Comment by pkgw** [comment]
More progress. It looks like the data coming off of `www.` are sometimes OK, sometimes bad. Right now, the following URL is reliably bad, because a bad image has been cached by the CDN:

```
http://cdn.worldwidetelescope.org/wwtweb/MarsMoc.aspx?q=2,0,1&v=2
```

This URL is sometimes OK, sometimes busted. I had to reload it about 5 times to get a bad version:

```
http://www.worldwidetelescope.org/wwtweb/MarsMoc.aspx?q=2,0,1&v=2
```

So now I'm guessing some kind of async problem in the MOC provider. @twsouthwick any guesses?

**Comment by twsouthwick** [comment]
Weird. I'll take a look

**Comment by twsouthwick** [comment]
I'm not able to repro it on Window, but I can on Linux. It appears there are a number of compat issues with linux.

**Comment by twsouthwick** [comment]
Looks like this may be a reason to investigate moving to https://github.com/SixLabors/ImageSharp sooner than later

**Comment by twsouthwick** [comment]
I've created an issue at dotnet/runtime to track root-causing this: https://github.com/dotnet/runtime/issues/47720

**Comment by pkgw** [comment]
Thanks for investigating! Maybe it will quickly become clear what's going on in your issue, but if not, it might be good to try porting this endpoint over to ImageSharp as a first move towards migrating more generally.

**Comment by pkgw** [comment]
Hmm, from the posted link to [this comment](https://github.com/dotnet/runtime/issues/47122#issuecomment-770056183) I get the impression that we're not going to see any action on this issue at deeper layers. @twsouthwick do you think you have the bandwidth to investigate ImageSharp for this endpoint? I don't have a sense as to how much of an effort that might be.

**Comment by pkgw** [comment]
I believe that I've fixed this in #283. It was pretty straightforward to swap out System.Drawing with SixLabors.ImageSharp ... at least I haven't detected any issues yet.
